lib/script.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable */
+'use strict';
+
+/**
+ * Validation for File Input Plugin
+ *
+ * @param {Object} input - inputs in this plugin's instance and their values as {String}
+ * @param {Object} input.target - the triggering input's name and value as {String}
+ * @param {String} input.target.name - the triggering input's name
+ * @param {String|Bool} input.target.value - the triggering input's value
+ * @param {Object} input.all - all inputs in this plugin's instance
+ * @param {String|Bool} input.all.selectsRelated - value of input selectsRelated
+ * @param {Object} [settings] - settings for this input plugin instance
+ * @param {Object} [settings.target] - the triggering input's settings as an {Object}
+ * @param {Object} [settings.all] - all settings in this plugin instance as an {Object}
+ * @param {Object} [settings.all.selectsRelated] - settings of input selectsRelated {Object}
+ *
+ * @returns {Bool|String} true or a string with the text describing the error
+ *
+ * @module fileScript
+ */
+
+module.exports = function fileScript(input, settings) {
+  const checkbox = input.all.delete;
+  const upload = input.all.file;
+  const container = input.parent.parentNode;
+  const link = container.querySelector('.file--link');
+  const deleter = container.querySelector('.file--delete');
+  let file;
+
+  /**
+   * Load handler for FileReader
+   * @param  {[type]} event - handler executed when the load event is fired
+   */
+  const loadHandler = function loadHandler (event) {
+    // change the link to the new file
+    link.setAttribute('href', event.target.result);
+
+    // make sure checkbox is no longer checked
+    checkbox.checked = false;
+
+    // make link not hidden
+    link.style.display = '';
+
+    // make delete checkbox not hidden
+    deleter.style.display = '';
+
+    // image uploaded
+    if (file.type.match('image.*')) {
+      // create image
+      const image = new Image();
+      image.title = file.name;
+      image.src = event.target.result;
+
+      // replace file--link contents
+      link.innerHTML = '';
+      link.appendChild(image);
+    }
+    else {
+      // non-image file
+      link.innerHTML = file.name;
+    }
+  }
+
+  /**
+   * Upload handler for file upload element
+   * @param  {[type]} event - handler executed when the load event is fired
+   */
+  const uploadHandler = function uploadHandler (event) {
+    if (event.target.files && event.target.files[0]) {
```

Comment:
Instead of repeating `event.target`, why not store it as `const target = event.target`?